fontir/src/depgraph.rs

```diff
@@ -227,23 +306,41 @@ impl DepGraph {
         new_files.iter().for_each(|p| paths.push(p));
 
         // Anything change 'round here?
+        let mut path_states = HashMap::new();
+        let mut changed = HashSet::new();
         for path in paths {
             let path_state = PathState::of(path)?;
             let has_changed = self
                 .entries
                 .get(path)
                 .map(|prior_state| *prior_state != path_state)
                 .unwrap_or(true);
+            path_states.insert(path, path_state);
 
             if has_changed {
-                changes.push(Change {
-                    path: path.to_owned(),
-                    path_state,
-                });
+                // Add path and, recursively, things that depend on it
+                let mut to_add = Vec::new();
+                to_add.push(path);
+                while let Some(path) = to_add.pop() {
+                    if !changed.insert(path) {
+                        continue;
+                    }
+                    if let Some(deps) = self.dependencies.get(path) {
+                        deps.iter().for_each(|e| to_add.push(e));
```

Comment:
I would write this as,
```suggestion
                        to_add.extend(deps.iter());
```

or maybe even something like,
```suggestion
                    to_add.extend(self.dependencies.get(path).into_iter().flat_map(Vec::iter));
```

although that might be a bit obscure. (Also I haven't checked that this compiles)

Reply:
I went with `to_add.extend(deps.iter());`, terse and simple ftw